drivers/ethernet/eth_liteeth.c

```diff
@@ -208,8 +218,11 @@ static void eth_iface_init(struct net_if *iface)
 #endif
 
 	/* set MAC address */
-	net_if_set_link_addr(iface, context->mac_addr, sizeof(context->mac_addr),
-			     NET_LINK_ETHERNET);
+	if(net_if_set_link_addr(iface, context->mac_addr, sizeof(context->mac_addr),
+			     NET_LINK_ETHERNET) < 0) {
+		LOG_ERR("setting mac failed");
+		return;	
+	}
```

Comment:
This looks like it is not part of "TX ready" change. Could you create a separate commit for this.